client/allocrunner/taskrunner/task_runner.go

```diff
@@ -1144,6 +1144,7 @@ func (tr *TaskRunner) buildTaskConfig() *drivers.TaskConfig {
 		Namespace:     alloc.Namespace,
 		NodeName:      alloc.NodeName,
 		NodeID:        alloc.NodeID,
+		ParentJobId:   alloc.Job.ParentID,
```

Comment:
Nitpicky golang style thing:

```suggestion
		ParentJobID:   alloc.Job.ParentID,
```